### ISSUE TITLE
False warning "@param documentation sections but no arguments" on using declarations/typedefs

### ISSUE THREAD

**Issue by Optimierungswerfer** [doxygen/doxygen#9552]
**Describe the bug**
A false warning

> Example.h:3 warning: FunctionSignature has @param documentation sections but no arguments

is being raised. It is false, because the function signature below clearly has arguments. This happens only with function signatures in typedefs/using declarations and not with "regular" function declarations (see example below).
The output HTML still lists the whole function signature under typedefs as well as the respective param documentation (with the exception that the argument of the typedef/using lacks the syntax highlighting of the argument in the regular function).

**Expected behavior**
No warning being raised.

**To Reproduce**
The example below illustrates the issue. [EmptyArgumentListIssueExample.zip](https://github.com/doxygen/doxygen/files/9445532/EmptyArgumentListIssueExample.zip) contains the Doxyfile that was used and this example code.
```h
// Example.h

/**
* A using declaration of a function signature with a parameter raises a warning about trying to document an argument when there are no arguments in the signature allegedly.
* @param argument I'm valid! :(
*/
using FunctionSignature = void (*) (int argument /* notice me :c */);

/**
* In a regular function declaration Doxygen detects the argument list perfectly fine and raises no warning. 
* @param argument I'm valid! :)
*/
void functionDeclaration(int argument);
```

**Version**
1.9.5 (2f6875a5ca481a69a6f32650c77a667f87d25e88) on Windows 10 (64 Bit)

**Additional context**
Pull Request #9496 added this [warning](https://github.com/doxygen/doxygen/pull/9496/commits/14a1de12affc2798278b7755a70d285c3194ee7f#diff-1535b42f903b8862efe99a9062a800fe10b2a198d57da8e6a5de43688958c543R382) in case the [argument list IS empty](https://github.com/doxygen/doxygen/blob/master/src/docparser.cpp#L317) (-> else-block), so that is where the symptom originates from. The underlying issue, however, must be deeper somewhere in the evaluation of argument lists, because in the attached example, the argument list should not be empty and thus the else-block that handles empty argument lists should not have been entered in the first place.

**Comment by IvanoBilenchi** [comment]
Same warning is raised when documenting function pointer fields. The warning is not raised in version 1.9.4.

```c
typedef struct MyStruct {
    /**
     * This raises a warning.
     * @param arg An argument.
     */
    void (*my_function)(void *arg);
} MyStruct;
```

**Comment by doxygen** [comment]
Indeed, the warning that was introduced is too simple/naive and does not take function pointers or function types into account. 

You can of course argue that a pointer or type itself has no parameters, only the thing it refers to has, but this doesn't mean it should not be possible to document the parameters anyway.

**Comment by doxygen** [comment]
I've limited the parameter check to real functions now. Could still be improved, but at least this should prevent the "false" warnings when `@param` is used on other types.

**Comment by StarFire75** [comment]
I have the same problem with getters for indexed properties in IDL files, e.g.

`    /** \brief Returns the filter settings for the given index
      *
      * Returns the filter settings for the given index. Throws an exception if the
      * index is not valid.
      *
      * \param[in] p_Index Zero based filter index
      *
      * \see SettingsByType
      */
    [propget,id(0x00000002)] HRESULT _stdcall SettingsByIndex([in] int p_Index, [out, retval] IFilterSettings** p_Value);
`

causes it. Does your fix cover this case too?

**Comment by albert-github** [comment]
Did you try it?

With the doxygen version 1.9.5 I gwet the warning:
```
warning: SettingsByIndex has @param documentation sections but no arguments
```
with the current master (1.9.6 (c345c22aa4d54de4a752e03a5fc976bae9f08fdd)) the warning is not given anymore.

**Comment by lukech** [comment]
I wonder if there's any workaround without the need to upgrade the Doxygen version to 1.9.6 (or a newer version with the fix)?

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.6.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
